lib/install/management/store_files.go

```diff
@@ -106,16 +106,62 @@ func (d *Dispatcher) deleteDatastoreFiles(ds *object.Datastore, path string, for
 		log.Debugf("Folder %q is not empty, leave it there", dsPath)
 		return empty, nil
 	}
-	if err = d.deleteVMFSFiles(ds, dsPath, d.force); err == nil {
-		empty = true
+
+	m := object.NewFileManager(ds.Client())
+	if d.isVSAN(ds) {
+		if err = d.deleteFilesRecursively(m, ds, dsPath); err != nil {
+			return empty, err
+		}
+		return true, nil
 	}
-	return empty, err
+
+	if err = d.deleteVMFSFiles(m, ds, dsPath); err != nil {
+		// let's try to remove recursively here, even if it's not VSAN
+		if nerr := d.deleteFilesRecursively(m, ds, dsPath); nerr != nil {
+			log.Debugf("Recursively delete failed as well: %s", nerr)
+			return empty, err
+		}
+		log.Infof("Datastore file %q deleted", dsPath)
+		return true, nil
+	}
+	return true, nil
+}
+
+func (d *Dispatcher) isVSAN(ds *object.Datastore) bool {
+	dsType, _ := ds.Type(d.ctx)
+
+	return dsType == types.HostFileSystemVolumeFileSystemTypeVsan
 }
 
-func (d *Dispatcher) deleteVMFSFiles(ds *object.Datastore, dsPath string, force bool) error {
+func (d *Dispatcher) deleteFilesRecursively(m *object.FileManager, ds *object.Datastore, dsPath string) error {
```

Comment:
Unfortunately, it does not work with non VSAN. So I removed the additional attempt if it failed in normal datastore.